dune

```diff
@@ -0,0 +1,9 @@
+(subdir run
```

Comment:
I think this should be called `runtest`.

Alternatively now that I understand a bit better what is going on, moving this back to `tests` should work, no?
(I think this is your original solution but w/o the `input` directory)

```dune
(subdir runtest
  (dynamic_include ../gentest/dune.inc))

(subdir gentest
  (rule
    (deps (source_tree ..))
    (action
      (with-stdout-to dune.inc (run ../../tools/gentest.exe ..)))))
```

Reply:
I am 99% sure it will not work but okay I can try again... I am sure this was my first version actually.

Reply:
Uhm, actually,  if you put the dune.inc in a different directory, It may work!

Reply:
As expected:
```
Error: Dependency cycle between:
   dynamic_include tests/gentest/dune.inc in directory tests/runtest
-> { dir = In_build_dir "default/tests/runtest"
   ; predicate = False
   ; only_generated_files = false
   }
-> _build/default/tests/gentest/dune.inc
-> dynamic_include tests/gentest/dune.inc in directory tests/runtest
```

Reply:
Ah duh the `source_tree ..` yeah; it'd have to be something like `(deps (glob_files_rec ../*.ae) (glob_files_rec ../*.smt2) (glob_files_rec ../*.psmt2))` or something. I'm OK with keeping this at toplevel with a `source_tree` then.

Reply:
Yep, I tried the solution with `glob_files_rec` too but having the rule at toplevel is fine.